controller/snapshot_controller.go

```diff
@@ -299,15 +300,17 @@ func (sc *SnapshotController) reconcile(snapshotName string) (err error) {
 		if err != nil && !shouldUpdateObject(err) {
 			return
 		}
-		if reflect.DeepEqual(existingSnapshot.Status, snapshot.Status) {
-			return
-		}
-
-		snapshot, err = sc.ds.UpdateSnapshotStatus(snapshot)
-		if err != nil {
-			return
+		if !reflect.DeepEqual(existingSnapshot.Annotations, snapshot.Annotations) {
+			// We need to annotate this deleting snapshot.
+			if snapshot, err = sc.ds.UpdateSnapshot(snapshot); err != nil {
+				return
+			}
```

Comment:
NIT: Can we update the CR spec/meta inside its own controller?

Reply:
I discussed this with @PhanLe1010 and @james-munson as well, but am waiting for their review to see what the consensus is. The other option is a `snapshot.status.attachingForDeletion` field. This more correctly adheres to our standard (a controller can only update its own status and other resource specs). I feel that the annotation is more natural than a status field, but will switch if that's what the team wants.